bridges/otelzap/core_test.go

```diff
@@ -53,3 +62,45 @@ func TestNewCoreConfiguration(t *testing.T) {
 		assert.Equal(t, want, got)
 	})
 }
+
+// Test conversion of Zap Level to OTel level.
+func TestGetOTelLevel(t *testing.T) {
+	tests := []struct {
+		level       zapcore.Level
+		expectedSev log.Severity
+	}{
+		{zapcore.DebugLevel, log.SeverityDebug},
+		{zapcore.InfoLevel, log.SeverityInfo},
+		{zapcore.WarnLevel, log.SeverityWarn},
+		{zapcore.ErrorLevel, log.SeverityError},
+		{zapcore.DPanicLevel, log.SeverityFatal1},
+		{zapcore.PanicLevel, log.SeverityFatal2},
+		{zapcore.FatalLevel, log.SeverityFatal3},
+		{zapcore.InvalidLevel, log.SeverityUndefined},
+	}
+
+	for _, test := range tests {
+		result := getOTelLevel(test.level)
+		if result != test.expectedSev {
+			t.Errorf("For level %v, expected %v but got %v", test.level, test.expectedSev, result)
+		}
+	}
+}
+
+// Tests [Core] write method.
+func TestCore(t *testing.T) {
+	rec := logtest.NewRecorder()
+	zc := NewCore(WithLoggerProvider(rec))
+
+	t.Run("test Write method of Core", func(t *testing.T) {
+		logger := zap.New(zc)
+		logger.Info(testBodyString, testField)
+
+		// not sure index 1 populated with results and not 0
```

Comment:
Thanks for this comment. This is worth checking. I can try to take a look at it when I have some time.